docs/_posts/2022-05-06-writing-cli-tools-with-typescript.md

```diff
@@ -0,0 +1,187 @@
+
+HEY!
+```
+
+We can also add a quick test by adding *src/yell.test.ts* with the following content
+
+```typescript
+import buildCmd from "./cmd";
+
+beforeEach(() => {
+ jest.spyOn(process, "exit").mockImplementation();
+ jest.spyOn(console, "log").mockImplementation();
+});
+
+describe("Yell subcommand", () => {
+ it("Should say 'HEY!' when passed 'hey!'", () => {
+  const cmd = buildCmd();
+  cmd.parse(["node", "cmd", "yell", "hey!"]);
+
+  expect(console.log).toHaveBeenCalledWith("HEY!");
+ });
+});
+```
+
+Since our Jest configuration expects JavaScript test files, we first have to transpile our code with the build script using the TypeScript compiler *tsc*. The transpiled JavaScript source code is put into the dist/ folder.
+
+```bash
+npm run build
+> @fkurz/typescript-cliutil-starter@1.0.0 build /private/tmp/my-cli
+> tsc --build
+```
+
+
+
+We can now run our test suite with Jest using the test script. New test files will automatically get recognized by Jest if they have—as in our case—the suffix *.test.js*
+
+```bash
+$ npm run test
+> @fkurz/typescript-cliutil-starter@1.0.0 test /private/tmp/my-cli
+> jest
+
+...
+
+Test Suites: 3 passed, 3 total
+Tests:       8 passed, 8 total
+Snapshots:   0 total
+Time:        0.41 s, estimated 1 s
+Ran all test suites.
+```
+As a final step, we can build a native binary by executing npm run package. The pkg tool will then bundle all source code and dependencies into a single binary file *bin/main*. 
+
+```bash
+$ npm run package
+
+> @fkurz/typescript-cliutil-starter@1.0.0 package /private/tmp/my-cli
+> pkg dist/main.js --no-bytecode --public-packages '*' --public --target host --output bin/main
+
+> pkg@5.3.1
+```
+
+Before running the program, we have to make it executable 
+
+```bash
+chmod +x ./bin/main
+```
+Now we can simply run the binary from the shell without a Node.js installation without any dependency installations whatsoever.
+
+```bash
+$ ./bin/main yell 'hey!'
+HEY!
+```
+
+## Summary
+
+TypeScript is a good choice for writing a CLI utility program due to various factors including being a developer-friendly language that also has static checks to assure code quality and because it offers a wide range of tooling choices for command-line programs within its ecosystem.
+
+Moreover, as shown above, developing a CLI utility written in TypeScript does not preclude the option of exporting a stand-alone, native binary if we use a tool like pkg to bundle our program and dependencies into an application binary.
+
+## About the Author: Friedrich Kurz
+
+I have been working for MaibornWolff as a full-time software engineer for 2.5 years now. In my current project, I'm working on AWS cloud infrastructure development for a client's web platform. I would describe myself as a technology generalist with a broad range of interests rather than a technology specialist but I'm also very interested in functional programming and general methods of writing correct, clean, and maintainable code.
```

Comment:
It may feel odd, but other _About the Author_'s have been written in 3rd person. Thus...

> Friedrich has been working for MaibornWolff...